### PR TITLE
Adds Unit Test for AWS Custom Logs

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/18595|

## Description

Adds unit test for the AWS Custom Logs.

### CODE REVIEW

File: wodles/aws/tests/test_s3_log_handler.py [hunk 1]
```diff
@@ -91,3 +92,51 @@ def test_aws_sl_subscriber_bucket_process_file(mock_wazuh_integration, mock_sts_
...
+    aws_ssl_subscriber_bucket = utils.get_mocked_aws_sl_subscriber_bucket()
+    message_body = {'bucket_path': 'example-bucket', 'log_path': 'example-log.parquet'}
+    mocked_obtain_logs = MagicMock(return_value=content)
+    aws_ssl_subscriber_bucket.obtain_logs = mocked_obtain_logs
+
+    with patch.object(aws_ssl_subscriber_bucket, 'send_msg') as mock_send_msg:
+        aws_ssl_subscriber_bucket.process_file(message_body)
+
+    mocked_obtain_logs.assert_called_once_with(bucket='example-bucket', log_path='example-log.parquet')
+    mock_send_msg.assert_has_calls(expected_calls)
```
Comment (GGP1) [inline wodles/aws/tests/test_s3_log_handler.py:142]: Add end of file line

File: wodles/aws/tests/test_s3_log_handler.py [hunk 2]
```diff
@@ -91,3 +92,51 @@ def test_aws_sl_subscriber_bucket_process_file(mock_wazuh_integration, mock_sts_
...
+    """Test the 'is_csv' function of AWSSubscriberBucket class."""
+    valid_csv_file = io.StringIO(content)
+    assert s3_log_handler.AWSSubscriberBucket.is_csv(valid_csv_file) == is_csv
+
+@pytest.mark.parametrize("content, expected",
+                         [
+                             (['{"event1": "data1"}', '{"event2": "data2"}'],
+                              [{'event1': 'data1'}, {'event2': 'data2'}])
+                         ])
+def test_process_jsonl(content, expected):
```
Comment (GGP1) [inline wodles/aws/tests/test_s3_log_handler.py:113]:
```suggestion
def test_protected_process_jsonl(content, expected):
```

File: wodles/aws/tests/test_s3_log_handler.py [hunk 3]
```diff
@@ -91,3 +92,51 @@ def test_aws_sl_subscriber_bucket_process_file(mock_wazuh_integration, mock_sts_
...
+    """Test the '_process_jsonl' function of AWSSubscriberBucket class."""
+    assert s3_log_handler.AWSSubscriberBucket._process_jsonl(content) == expected
+
+
+@pytest.mark.parametrize("content, expected", [
+    ({'example1': None, 'example2': 'some_value'}, {'example2': 'some_value'}),
+    ({'example1': {'a': None, 'b': None}, 'example2': {'a': 1, 'b': None}},
+     {'example1': {}, 'example2': {'a': 1}})
+])
+def test_remove_none_fields(content, expected):
```
Comment (GGP1) [inline wodles/aws/tests/test_s3_log_handler.py:123]:
```suggestion
def test_protected_remove_none_fields(content, expected):
```

File: wodles/aws/tests/test_sqs_message_processor.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        ),
+    ],
+)
+def test_extract_message_info(sqs_processor, sqs_messages, expected):
+    """
+    Test extract_message_info method of AWSQueueMessageProcessor subclasses.
+    """
+    processor = sqs_processor()
+    result = processor.extract_message_info(sqs_messages)
+    assert result == expected
```
Comment (GGP1) [inline wodles/aws/tests/test_sqs_message_processor.py:180]: I would prefer to do the `json.dumps` inside the function. Something like

```python
def test_extract_message_info(sqs_processor, sqs_messages, expected):
    """
    Test extract_message_info method of AWSQueueMessageProcessor subclasses.
    """
    processor = sqs_processor()
    for i, message in sqs_messages:
        sqs_messages[i]['Body'] = json.dumps(message['Body'])
    result = processor.extract_message_info(sqs_messages)
    assert result == expected
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: wodles/aws/tests/test_sqs_message_processor.py (hunk 4 above)
Comment (RamosFe) [reply wodles/aws/tests/test_sqs_message_processor.py]: https://github.com/wazuh/wazuh/blob/8f8f66ff33b8f07634b7dfe296be74d9ef01d703/wodles/aws/tests/test_sqs_message_processor.py#L165-L173

File: wodles/aws/tests/test_s3_log_handler.py (hunk 2 above)
Comment (RamosFe) [reply wodles/aws/tests/test_s3_log_handler.py:113]: https://github.com/wazuh/wazuh/blob/8f8f66ff33b8f07634b7dfe296be74d9ef01d703/wodles/aws/tests/test_s3_log_handler.py#L131

File: wodles/aws/tests/test_s3_log_handler.py (hunk 3 above)
Comment (RamosFe) [reply wodles/aws/tests/test_s3_log_handler.py]: https://github.com/wazuh/wazuh/blob/8f8f66ff33b8f07634b7dfe296be74d9ef01d703/wodles/aws/tests/test_s3_log_handler.py#L123

**Review by GGP1 — CHANGES_REQUESTED** [review]
Changes LGTM, but we should block the issue until all checks pass

**Review by GGP1 — APPROVED** [review]

**Pull request merged**